### PR TITLE
feat: getTech now accepts `TitleCase` or `camelCase`

### LINKED ISSUE #3986: Make TechOrder accept `camelCase` and `TitleCase` tech names
## Description
Currently `techOrder` only accepts tech names that are `camelCase` others will be ignored. It would be much more user friendly and closer to other conventions if we accept `TitleCase` as well.

## Proposed Changes
* After #3985 is merged remove the `camelCase` conversion.
* Anywhere that we use techOrder should search for tech using `camelCase` or `TitleCase`. Most of there are in `player.js`

### PR DESCRIPTION
**Pull request by BrandonOCasey**
## Description
`getTech` now works with `TitleCase` and `camelCase` tech names

Fixes #3986
## Specific Changes proposed
* update techOptions to look for `TitleCase`/`camelCase` options
* remove all possible `toTitleCase` calls from player.js
* remove deprecated usage of Tech as Component
* add a unit test to verify that registerTech works
* change defaultTech_ to defaultTechOrder_

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [x] Unit Tests updated or fixed
  - [ ] Docs/guides updated
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/js/player.js [hunk 1]
```diff
@@ -840,6 +843,8 @@ class Player extends Component {
       techOptions[props.getterName] = this[props.privateName];
     });
 
+    assign(techOptions, this.options_[titleTechName]);
```
Comment (gkatsev) [inline src/js/player.js]: does `assign` handle undefineds?

File: src/js/player.js [hunk 2]
```diff
@@ -851,14 +856,9 @@ class Player extends Component {
     }
 
     // Initialize tech instance
-    let TechComponent = Tech.getTech(techName);
+    const TechClass = Tech.getTech(techName);
 
-    // Support old behavior of techs being registered as components.
-    // Remove once that deprecated behavior is removed.
-    if (!TechComponent) {
-      TechComponent = Component.getComponent(techName);
```
Comment (gkatsev) [inline src/js/player.js]: I guess we should throw an error in this case?

Comment (BrandonOCasey) [inline src/js/player.js]: I will improve the error handing here

Comment (BrandonOCasey) [inline src/js/player.js]: I don't know if we show check `getComponent` at all. I think we should just throw an error if `TechComponent`/`TechClass` does not exist. We could check `getComponent` here but I think that we should be able to have a Tech and a Component with the same name (as weird as that would be)

File: src/js/player.js (hunk 1 above)
Comment (BrandonOCasey) [inline src/js/player.js]: Going to add some unit tests to assign to find out, will open another PR with those

File: src/js/player.js (hunk 2 above)
Comment (gkatsev) [inline src/js/player.js]: yeah, I meant to just throw if `TechClass` isn't available. We shouldn't try to call `getComponent` anymore.

File: src/js/player.js (hunk 1 above)
Comment (BrandonOCasey) [inline src/js/player.js]: See #4014 for tests, seems that undefined/null/empty will do nothing

File: src/js/player.js (hunk 2 above)
Comment (BrandonOCasey) [inline src/js/player.js]: Done

**Pull request merged**